kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -34,7 +34,7 @@
 @Evolving
 public interface TransactionBuilder {
   /**
-   * Set the schema of the table when creating a new table.
```

Comment:
Just calling out that this is the fundamental API change @allisonport-db @scottsand-db , we're leveraging the existing `withSchema` and loosening the underlying implementation to allow setting the schema for an existing table. IMO this is the best way for enabling a `setSchema` approach. Other approaches would needlessly widen the API surface area, but let me know what you think @allisonport-db @scottsand-db

Reply:
I think this is good. It makes sense, and this is marked as `@Evolving`.

I would however like a more complete javadoc the explains a bit what checks are done in the case that you're setting the schema on a new table (or links to docs to that effect)

Reply:
Updated with some more details on which conditions schema evolution can be performed and the expected validations implementations should perform